mlx-nn/src/convolution.rs

```diff
@@ -0,0 +1,456 @@
+use mlx_macros::ModuleParameters;
+use mlx_rs::module::{Module, Param};
+use mlx_rs::{
+    error::Exception,
+    ops::{conv1d, conv2d, zeros},
+    random::uniform,
+    Array,
+};
+
+use crate::utils::{IntOrPair, IntOrTriple, WithBias};
+
+/// Optional parameters for the `Conv1d` module.
+#[derive(Debug, Clone, Default)]
+pub struct Conv1dBuilder {
```

Comment:
@dcvz What about this kind of builder pattern? We could get rid of the `WithBias` type this way, and we could probably apply this on all `Module`'s that take optional args? For `cross_entropy`, we could make it a struct and then apply this approach?

Reply:
Or we could have two different way of handling optional args? Use this builder pattern on all `Module` structs, and keep functions like `cross_entropy` as it is?